onnxruntime/core/providers/dnnl/dnnl_execution_provider.cc

```diff
@@ -58,12 +66,41 @@ DNNLExecutionProvider::DNNLExecutionProvider(const DNNLExecutionProviderInfo& in
   if (!fusion_env.empty()) {
     enable_fusion_ = (std::stoi(fusion_env) == 0 ? false : true);
   }
+
+  // Set the number of threads specified by the user
+  // If provided arguments set them as the number of threads, else call
+  // calc which usually = numcores
+  auto num_threads = static_cast<int*>(info.threadpool_args);
+#if defined(DNNL_OPENMP)
```

Comment:
Yes

Reply:
Hmm, ok, then how does this address over subscription? Do you plan to document that users using onednn should set ORT num threads to 1 to avoid unnecessary threads (obviously this assumes all ops will get executed by onednn or else the ops that run by ORT CPU EP will only get 1 thread to run affecting perf)?

Reply:
We solve this by reducing the default number of threads OpenMP uses or just selecting an user defined number of threads (for fine tuning). Users can tune the number of ORT threads but is not mandatory.